src/worker/tasks/sendTransactionWorker.ts

```diff
@@ -427,16 +433,25 @@ export const getPopulatedOrErroredTransaction = async (
     if (!from) throw new Error("Invalid transaction parameters: from");
     if (!to) throw new Error("Invalid transaction parameters: to");
 
+    // Add extraGas to the gas limit if not explicitly set.
+    let extraGas = 0n;
+    if (!queuedTransaction.gas) {
+      extraGas = (await ConfigurationDB.getAll()).extraGas;
+    }
+
+    const preparedTransaction = prepareTransaction({
+      client: thirdwebClient,
+      chain: await getChain(queuedTransaction.chainId),
+      ...queuedTransaction,
+      to: getChecksumAddress(to),
+      // if transaction is EOA, stub the nonce to reduce RPC calls
+      nonce: queuedTransaction.isUserOp ? undefined : 1,
+      extraGas,
```

Comment:
i was kinda expecting this to be added per request, not as a config flag for the whole engine instance.

why isnt extraGas part of the txOverrides ?

Reply:
Hmmm we could do that instead. Not sure if the need for this will commonly be for a chain or use case where the gas estimate is always too low, hence the global setting. 

A per-tx setting is more flexible but requires a code change on the dev side. If it's likely a dev would want to add gas only for certain requests, this makes more sense.

Reply:
From experience this is on a per request basis. You only need this for calls that have some random execution or chain of delegate calls.

But doesn't make sense to add it for a basic transfer. 

Thinking about ecosystem engine instances that can be shared, def doesn't make sense to have this as a global flag